clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractCoordinator.java

```diff
@@ -835,6 +839,8 @@ public void handle(SyncGroupResponse syncResponse,
                 } else if (error == Errors.REBALANCE_IN_PROGRESS) {
                     log.info("SyncGroup failed: The group began another rebalance. Need to re-join the group. " +
                                  "Sent generation was {}", sentGeneration);
+                    savePartitionAndGenerationState();
```

Comment:
saving the generation and partition state prior to resetting it.

Reply:
couldn’t we just rejoin without reseting the generation? in other words, what is the advantage of resetting the generation here if we still want to use it to re-join afterwards?

Reply:
I'm not sure what's the better way to invoke `onPartitionsLost` w/o resetting the generation. The alternative is to use a flag, but that's rather unclean.

Reply:
This is discarded in the latest commit